-Wnarrowing warning on 32-bit system

On Debian 10.10 32-bit with GCC 8.3.0:
```
...
bench/peer_eviction.cpp: In function ‘void EvictionProtectionCommon(ankerl::nanobench::Bench&, int, std::function<void(NodeEvictionCandidate&)>)’:
bench/peer_eviction.cpp:30:51: warning: narrowing conversion of ‘(((uint64_t)(& bench)->ankerl::nanobench::Bench::epochs()) * (& bench)->ankerl::nanobench::Bench::epochIterations())’ from ‘uint64_t’ {aka ‘long long unsigned int’} to ‘std::vector<std::vector<NodeEvictionCandidate> >::size_type’ {aka ‘unsigned int’} inside { } [-Wnarrowing]
     std::vector<Candidates> copies{bench.epochs() * bench.epochIterations(), candidates};
                                    ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
...
```

`bench/peer_eviction.cpp` was added recently in #22284.

--- a/src/bench/peer_eviction.cpp
+++ b/src/bench/peer_eviction.cpp
@@ -27,7 +27,8 @@ static void EvictionProtectionCommon(
         candidate_setup_fn(c);
     }
 
-    std::vector<Candidates> copies{bench.epochs() * bench.epochIterations(), candidates};
+    std::vector<Candidates> copies{
+        static_cast<size_t>(bench.epochs() * bench.epochIterations()), candidates};
     size_t i{0};
     bench.run([&] {
         ProtectEvictionCandidatesByRatio(copies.at(i));

